pkg/activitypub/vocab/objectproperty.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package vocab
+
+import (
+	"encoding/json"
+	"net/url"
+)
+
+// ObjectProperty defines an 'object' property. The property may be a simple IRI or
+// an embedded object such as 'Collection', 'OrderedCollection', 'Activity', etc.
+type ObjectProperty struct {
+	iri *URLProperty
+	obj *ObjectType
+}
+
+// NewObjectProperty returns a new 'object' property with the given options.
+func NewObjectProperty(opts ...Opt) *ObjectProperty {
+	options := NewOptions(opts...)
+
+	return &ObjectProperty{
+		iri: NewURLProperty(options.Iri),
+		obj: options.Object,
+	}
+}
+
+// GetType returns the type of the object property. If the property
```

Comment:
I usually think that Get prefix should be omitted for "local" object properties (e.g., that don't need a fetch from a resource).

Reply:
Yeah, I think I used GetXXX for properties early on since I had naming conflicts with public properties that were being marshalled. But I can change it.

Reply:
I'll change it in a separate PR since I also used GetXXX in the previous PR.